src/main/java/org/jilt/internal/AbstractBuilderGenerator.java

```diff
@@ -338,15 +340,17 @@ protected final ParameterSpec setterParameterSpec(VariableElement attribute, Typ
         ParameterSpec.Builder ret = ParameterSpec
                 .builder(parameterType, this.attributeSimpleName(attribute))
                 .addModifiers(Modifier.FINAL);
-        this.addAnnotationsToParam(ret, attribute.getAnnotationMirrors());
-        this.addAnnotationsToParam(ret, attribute.asType().getAnnotationMirrors());
+        Set<String> presentAnnotations = new HashSet<String>();
+        this.addAnnotationsToParam(ret, attribute.getAnnotationMirrors(), presentAnnotations);
+        this.addAnnotationsToParam(ret, attribute.asType().getAnnotationMirrors(), presentAnnotations);
         return ret.build();
     }
 
-    private void addAnnotationsToParam(ParameterSpec.Builder param, List<? extends AnnotationMirror> annotations) {
+    private void addAnnotationsToParam(ParameterSpec.Builder param, List<? extends AnnotationMirror> annotations, Set<String> presentAnnotations) {
         for (AnnotationMirror annotation : annotations) {
-            if (this.isAnnotationAllowedOnParam(annotation)) {
+            if (!presentAnnotations.contains(annotation.toString()) && this.isAnnotationAllowedOnParam(annotation)) {
                 param.addAnnotation(AnnotationSpec.get(annotation));
+                presentAnnotations.add(annotation.toString());
```

Comment:
What if the annotation is repeatable? In that case, should we repeat it?

Reply:
I've added a test to check this. It worked exactly the same as non-repeatable annotations, so they would be copied twice. I don't think this test is compatible, so I'll remove it if you want me to

Reply:
I don't really know why, if tests run on 8+...